st2common/st2common/util/schema/__init__.py

```diff
@@ -136,7 +137,9 @@ def assign_default_values(instance, schema):
         schema_items = property_data.get('items', {})
 
         # Array
-        if attribute_type == 'array' and schema_items and schema_items.get('properties', {}):
+        if (schema_items and schema_items.get('properties', {}) and
+                (attribute_type == 'array' or
+                    (isinstance(attribute_type, list) and 'array' in attribute_type))):
```

Comment:
Good catch with `list` - we don't actually allow user to provide multiple types for action parameters (so it's never a list), but there are probably some other places we control (e.g. API models perhaps?) where the type can be a list, right?